src/ua/griddynamics/Main.java

```diff
@@ -0,0 +1,131 @@
+            int choice = SCANNER.nextInt();
+            SCANNER.nextLine();
+            switch (RootMenu.getInstance(choice)) {
+                case ADD_INCOME:
+                    System.out.println("\nEnter income:");
+                    double income = SCANNER.nextDouble();
+                    SCANNER.nextLine();
+                    purchase.addIncome(income);
+                    System.out.println("Income was added!\n");
+                    break;
+                case ADD_PURCHASE:
+                    addPurchase(purchase);
+                    break;
+                case SHOW_LIST_OF_PURCHASES:
+                    while (true) {
+                        System.out.println(ShowListMenu.getMenuStr());
+                        int choiceForShowList = SCANNER.nextInt();
+                        SCANNER.nextLine();
+                        if (ShowListMenu.getInstance(choiceForShowList).equals(ShowListMenu.BACK)) {
+                            System.out.println();
+                            break;
+                        } else {
+                            System.out.println(purchase.showList(choiceForShowList));
+                        }
+                    }
+                    break;
+                case BALANCE:
+                    System.out.println(purchase.showBalance());
+                    break;
+                case SAVE:
+                    purchase.savePurchases();
+                    System.out.println("\nPurchases were saved!\n");
+                    break;
+                case LOAD:
+                    purchase.loadPurchase();
+                    System.out.println("\nPurchases were loaded!\n");
+                    break;
+                case ANALYZE:
+                    showSort(purchase);
+                    break;
+                case EXIT:
+                    quit = true;
+                    System.out.println("\nBye!");
+                    SCANNER.close();
+                    break;
+                case UNDEFINED:
+                    System.out.println("\nSomething went wrong.\n");
+            }
+        }
+    }
+
+    private static void addPurchase(Purchase purchase) {
+        String name = null;
+        double price = 0.0;
+        while (true) {
+            System.out.println(AddPurchaseMenu.getMenuStr());
+            int choice = SCANNER.nextInt();
+            SCANNER.nextLine();
+            if (!AddPurchaseMenu.getInstance(choice).equals(AddPurchaseMenu.BACK)) {
+                try {
+                    System.out.println("\nEnter purchase name:");
+                    name = SCANNER.nextLine();
+                    System.out.println("Enter its price:");
+                    price = SCANNER.nextDouble();
+                    SCANNER.nextLine();
+                } catch (InputMismatchException e) {
+                    e.printStackTrace();
+                }
+                if (purchase.addProduct(new Product(name, price, ProductType.getInstance(choice)))) {
```

Comment:
ProductType should not depend on choice. 
Instead of this add productType field to AddPurchaseMenu. And use this new field here